File: ABC/ABC355/D/main.cc
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;
using ll = long long;

int main() {
  int n;
  cin >> n;

  vector<pair<int, int>> vp;
  for (int i = 0; i < n; ++i) {
    int l, r;
    cin >> l >> r;
    vp.emplace_back(l, 0);
    vp.emplace_back(r, 1);
  }
  sort(begin(vp), end(vp));

  ll cur = 0, res = 0;
  for (auto &[_, v] : vp) {
    if (v) {
      --cur;
      res += cur;
    } else {
      ++cur;
    }
  }
  cout << res << endl;
  return 0;
}

File: ABC/ABC355/D/main.py
n = int(input())
li = []

for _ in range(n):
    l, r = map(int, input().split())
    li.append((l, 0))
    li.append((r, 1))

li = sorted(li)
res = 0
cur = 0
for _, v in li:
    if v:
        cur -= 1
        res += cur
    else:
        cur += 1

print(res)